src/main/java/com/synopsys/integration/detect/lifecycle/boot/DetectBoot.java

```diff
@@ -199,19 +205,39 @@ public Optional<DetectBootResult> boot(String detectVersion, String detectBuildD
         } catch (DetectUserFriendlyException e) {
             return Optional.of(DetectBootResult.exception(e, propertyConfiguration, directoryManager, diagnosticSystem));
         }
+        AutoScanTypeDecider autoDetectTool = new AutoScanTypeDecider();
+        Map<DetectTool, Set<String>> scanTypeEvidenceMap = autoDetectTool.decide(hasImageOrTar, detectConfiguration);
```

Comment:
Should this map be put inside Autonomous Manager class to pack and make it available in DetectRun stage?

Reply:
It should be part of one of the Detect***Configuration. That would fit the existing pattern.

Reply:
As discussed with Madhu, this will be done once code is available in the autonomous feature branch to avoid conflicts. Approving PR for now.

Reply:
Yeah I would agree with above for moving this entire code segment into a method in AutonomousManager in the interest of separation of concerns as we had discussed earlier. I see this branch is already up to date with feature/autonomous-scanning with no conflicts so can we make that change here itself? I feel it would be easier to build upon if the design is confirmed in this PR itself rather than creating another PR to move things around